test/functional/tests/cache_ops/test_cleaning_policy_operation.py

```diff
@@ -0,0 +1,261 @@
+                failed_params += (
+                    f"Flush Max Buffers is {current_alru_params.flush_max_buffers}, "
+                    f"should be {alru_params.flush_max_buffers}\n"
+                )
+            if current_alru_params.activity_threshold != alru_params.activity_threshold:
+                failed_params += (
+                    f"Activity Threshold is {current_alru_params.activity_threshold}, "
+                    f"should be {alru_params.activity_threshold}\n"
+                )
+            TestRun.fail(f"ALRU parameters did not switch properly:\n{failed_params}")
+
+    if cleaning_policy == CleaningPolicy.acp:
+        acp_params = FlushParametersAcp()
+        acp_params.wake_up_time = Time(milliseconds=100)
+        acp_params.flush_max_buffers = 32
+        cache.set_params_acp(acp_params)
+        current_acp_params = cache.get_flush_parameters_acp()
+        if current_acp_params != acp_params:
+            failed_params = ""
+            if current_acp_params.wake_up_time != acp_params.wake_up_time:
+                failed_params += (
+                    f"Wake Up time is {current_acp_params.wake_up_time}, "
+                    f"should be {acp_params.wake_up_time}\n"
+                )
+            if current_acp_params.flush_max_buffers != acp_params.flush_max_buffers:
+                failed_params += (
+                    f"Flush Max Buffers is {current_acp_params.flush_max_buffers}, "
+                    f"should be {acp_params.flush_max_buffers}\n"
+                )
+            TestRun.fail(f"ACP parameters did not switch properly:\n{failed_params}")
+
+
+def fio_prepare(core):
+    fio = (
+        Fio()
+        .create_command()
+        .io_engine(IoEngine.libaio)
+        .block_size(Size(4, Unit.KibiByte))
+        .size(io_size)
+        .read_write(ReadWrite.randwrite)
+        .target(core.system_path)
+        .direct(1)
+    )
+    return fio
+
+
+def check_cleaning_policy_operation(
+    cleaning_policy, core_writes_before_cleaning, core_writes_after_cleaning
+):
+    if cleaning_policy == CleaningPolicy.alru:
+        if (
+            core_writes_after_cleaning.value != io_size.value * cores_count
+            or core_writes_before_cleaning.value != 0
+        ):
+            TestRun.fail(
+                "ALRU cleaning policy is not working properly! "
+                "Cleaner should already finish flushing dirty data and "
+                "core writes should equal total IO on cache."
+            )
+    if cleaning_policy == CleaningPolicy.nop:
+        if core_writes_after_cleaning.value != 0 or core_writes_before_cleaning.value != 0:
+            TestRun.fail(
+                "NOP cleaning policy is not working properly! "
+                "There should be no core writes as there is no cleaning of dirty data."
+            )
+    if cleaning_policy == CleaningPolicy.acp:
+        if core_writes_after_cleaning <= core_writes_before_cleaning:
```

Comment:
I'd add a check here that core_writes_before_cleaning != 0
BTW if acp is cleaning all the time and nop is never cleaning, then before/after_cleaning are not exactly right names for variables ;)

Reply:
Done